tests/unit/src/main/scala/tests/BaseCodeLensLspSuite.scala

```diff
@@ -45,11 +47,28 @@ abstract class BaseCodeLensLspSuite(name: String) extends BaseLspSuite(name) {
               |$original
               |""".stripMargin
         )
+        _ <- extraInitialization(server, sourceFile)
         _ <- assertCodeLenses(sourceFile, expected, printCommand = printCommand)
       } yield ()
     }
   }
 
+  def checkTestCases(
+      name: TestOptions,
+      library: Option[String] = None,
+      scalaVersion: Option[String] = None,
+      printCommand: Boolean = false,
+  )(
+      expected: => String
+  )(implicit loc: Location): Unit = check(
+    name,
+    library,
+    scalaVersion,
+    printCommand,
+    (server, sourceFile) =>
+      server.discoverTestSuites(List(sourceFile)).map(_ => ()),
```

Comment:
A lot of flakyness without ensuring this was run before the assertions unfortunately